File: css/money/ch07/Money.Tests/FrancTest.cs
using NUnit.Framework;
using TheSoftwareGorilla.TDD.Money;

namespace TheSoftwareGorilla.TDD.Money.Tests;

public class FrancTests
{
    [SetUp]
    public void Setup()
    {
    }

    [Test]
    public void TestConstruction()
    {
        var five = new Franc(5);
        Assert.IsNotNull(five);
        Assert.That(five.Amount, Is.EqualTo(5));
        var ten = new Franc(10);
        Assert.IsNotNull(ten);
        Assert.That(ten.Amount, Is.EqualTo(10));
    }

    [Test]
    public void TestMultiplication()
    {
        Franc five = new Franc(5);
        Assert.That(new Franc(10), Is.EqualTo(five.Times(2)));
        Assert.That(new Franc(15), Is.EqualTo(five.Times(3)));
    }

}

File: java/money/ch08/src/test/java/com/thesoftwaregorilla/tdd/money/FrancTest.java
package com.thesoftwaregorilla.tdd.money;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class FrancTest {

    @Test
    public void testConstruction() {
        Money five = Money.franc(5);
        assertNotNull(five);
        assertEquals(5, five.getAmount());
        Money ten = Money.franc(10);
        assertNotNull(ten);
        assertEquals(10, ten.getAmount());
    }

    @Test
    public void testMultiplication() {
        Money five = Money.franc(5);
        assertEquals(Money.franc(10), five.times(2));
        assertEquals(Money.franc(15), five.times(3));
    }

}
